server/app/durablejobs/jobs/CalculateEligibilityDeterminationJob.java

```diff
@@ -72,11 +76,14 @@ public void run() {
                   "Calculating eligibility determination for application id {}",
                   application.id); // Use the captured ID
 
-              ProgramModel pm = application.getProgram();
-              ProgramDefinition programDefinition = pm.getProgramDefinition();
+              Long programId = application.getProgram().id;
+              ProgramDefinition programDefinition =
+                  programService.getFullProgramDefinition(programId);
```

Comment:
A note about testing: I added an "eligible" test to `CalculateEligibilityDeterminationTest` which duplicates most of [the ineligible test](https://github.com/civiform/civiform/blob/main/server/test/durablejobs/jobs/CalculateEligibilityDeterminationTest.java#L157), but both tests pass with/without this change so I didn't included it. It seems like those tests should fail with the same error we're seeing in prod, but they don't and I'm not sure why. I would love any advice so we can fully test this scenario!

Reply:
Great question, I'll try that. I'm pretty sure the check fails for all question types though.

Reply:
@dkatzz I wasn't able to get the test to fail using an enumerator question either. I'm going to resolve this for now and will look into these tests further later!